### PR TITLE
vim-patch:9.0.0935: when using dash it may not be recognize as filetype "sh"

### PR DESCRIPTION
**Pull request by clason**
Problem:    When using dash it may not be recognize as filetype "sh".
Solution:   Add checks for "dash". (Eisuke Kawashima,closes vim/vim#11600)

https://github.com/vim/vim/commit/24482fbfd599d2273c48951df7d00d62f3e66c85

Co-authored-by: Eisuke Kawashima <e-kwsm@users.noreply.github.com>

### CODE REVIEW

File: runtime/lua/vim/filetype/detect.lua [hunk 1]
```diff
@@ -1172,7 +1172,8 @@ function M.sh(path, contents, name)
       vim.b[b].is_kornshell = nil
       vim.b[b].is_sh = nil
     end
-  elseif matchregex(name, [[\<sh\>]]) then
+    -- Ubuntu links sh to dash
+  elseif matchregex(name, [[\<sh\>]]) or matchregex(name, [[\<dash>]]) then
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1176]:
```suggestion
  elseif matchregex(name, [[\<sh\>]]) or matchregex(name, [[\<dash\>]]) then
```

File: runtime/lua/vim/filetype/detect.lua [hunk 2]
```diff
@@ -1172,7 +1172,8 @@ function M.sh(path, contents, name)
       vim.b[b].is_kornshell = nil
       vim.b[b].is_sh = nil
     end
-  elseif matchregex(name, [[\<sh\>]]) then
+    -- Ubuntu links sh to dash
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1175]: I think this comment should be below the `elseif` and match the original comment in Vim better.

Comment (clason) [reply runtime/lua/vim/filetype/detect.lua:1175]: But that's how the comments are ordered in the rest of this file. Being inconsistent is even more confusing.

Comment (clason) [reply runtime/lua/vim/filetype/detect.lua:1175]: (And I don't think the Vim comment contains any more information; it is just longer. We port the logic, not the code.)

**Review by zeertzjq — APPROVED** [review]

File: runtime/lua/vim/filetype/detect.lua [hunk 3]
```diff
@@ -1172,7 +1172,8 @@ function M.sh(path, contents, name)
...
+  elseif matchregex(name, [[\<sh\>]]) or matchregex(name, [[\<dash\>]]) then
```
Comment (smjonas) [inline runtime/lua/vim/filetype/detect.lua:1176]: Maybe we can use `elseif matchregex(name, [[\<\(sh|dash\)\>]])` here

Comment (clason) [reply runtime/lua/vim/filetype/detect.lua:1176]: Probably want to do this a bit before for `bash` as well then?

File: runtime/lua/vim/filetype/detect.lua [hunk 4]
```diff
@@ -1166,13 +1166,14 @@ function M.sh(path, contents, name)
       vim.b[b].is_bash = nil
       vim.b[b].is_sh = nil
     end
-  elseif vim.g.bash_is_sh or matchregex(name, [[\<bash\>]]) or matchregex(name, [[\<bash2\>]]) then
+  elseif vim.g.bash_is_sh or matchregex(name, [[\<bash|bash2\>]]) then
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1169]:
```suggestion
  elseif vim.g.bash_is_sh or matchregex(name, [[\<(bash|bash2)\>]]) then
```

File: runtime/lua/vim/filetype/detect.lua [hunk 5]
```diff
@@ -1166,13 +1166,14 @@ function M.sh(path, contents, name)
...
     on_detect = function(b)
       vim.b[b].is_bash = 1
       vim.b[b].is_kornshell = nil
       vim.b[b].is_sh = nil
     end
-  elseif matchregex(name, [[\<sh\>]]) then
+    -- Ubuntu links sh to dash
+  elseif matchregex(name, [[\<sh|dash\>]]) then
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1176]:
```suggestion
  elseif matchregex(name, [[\<(sh|dash)\>]]) then
```

File: runtime/lua/vim/filetype/detect.lua (hunk 4 above)
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1169]:
```suggestion
  elseif vim.g.bash_is_sh or matchregex(name, [[\<\(bash|bash2\)\>]]) then
```

File: runtime/lua/vim/filetype/detect.lua (hunk 5 above)
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1176]:
```suggestion
  elseif matchregex(name, [[\<\(sh|dash\)\>]]) then
```

File: runtime/lua/vim/filetype/detect.lua (hunk 4 above)
Comment (clason) [reply runtime/lua/vim/filetype/detect.lua]: That teaches me to blindly rely on `make oldtest`...

**Review by zeertzjq — DISMISSED** [review]

File: runtime/lua/vim/filetype/detect.lua [hunk 6]
```diff
@@ -1166,13 +1166,14 @@ function M.sh(path, contents, name)
       vim.b[b].is_bash = nil
       vim.b[b].is_sh = nil
     end
-  elseif vim.g.bash_is_sh or matchregex(name, [[\<bash\>]]) or matchregex(name, [[\<bash2\>]]) then
+  elseif vim.g.bash_is_sh or matchregex(name, [[\<\(bash|bash2\)\>]]) then
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1169]:
```suggestion
  elseif vim.g.bash_is_sh or matchregex(name, [[\<\(bash\|bash2\)\>]]) then
```

File: runtime/lua/vim/filetype/detect.lua [hunk 7]
```diff
@@ -1166,13 +1166,14 @@ function M.sh(path, contents, name)
...
     on_detect = function(b)
       vim.b[b].is_bash = 1
       vim.b[b].is_kornshell = nil
       vim.b[b].is_sh = nil
     end
-  elseif matchregex(name, [[\<sh\>]]) then
+    -- Ubuntu links sh to dash
+  elseif matchregex(name, [[\<\(sh|dash\)\>]]) then
```
Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:1176]:
```suggestion
  elseif matchregex(name, [[\<\(sh\|dash\)\>]]) then
```

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**